<launch>

    <!-- Viso2 -->

	<!-- Arguments -->
    <arg name="camera" default="/head_camera_rgb"/><!--"/bottom_kinect/rgb"/--><!-- The namespace where images are published -->
    <param name="/use_sim_time" value="true"/>

	<!-- Run the ROS package stereo_image_proc -->
	<group ns="$(arg camera)" >
		<node pkg="image_proc" type="image_proc" name="image_proc"/>
    </group>

    <node pkg="viso2_ros" type="mono_odometer" name="mono_odometer" output="screen">
      <!--remap from="mono" to="$(arg camera)"/-->
      <remap from="image" to="$(arg camera)/image_raw"/>
      <remap from="/camera/camera_info" to="$(arg camera)/camera_info"/>

      <!--param name="base_link_frame_id" value="/base_link"/-->
      <param name="camera_height" value="0.890"/>
      <param name="camera_pitch" value="0.004"/>
      <!--param name="sensor_frame_id" value="/kinect_camera_rgb_optical_frame_bottom"/-->
    </node>

    <arg name="name"
         default="monocamera_visual_odometry_kalman_filter"  />
    <arg name="config_file"
         default="$(find monocamera_visual_odometry_kalman_filter)/config/params.yaml"  />

    <rosparam file="$(arg config_file)" command="load" ns="$(arg name)" />

    <node name="$(arg name)"
          pkg="monocamera_visual_odometry_kalman_filter"
          type="monocamera_visual_odometry_kalman_filter_node"
          output="screen"
          respawn="true" />
</launch>